packages/transactions/src/clarity/types/intCV.ts

```diff
@@ -1,41 +1,103 @@
-// @ts-ignore
 import { Buffer } from '@stacks/common';
 import BigNum from 'bn.js';
-import { CLARITY_INT_SIZE } from '../../constants';
+import { CLARITY_INT_BYTE_SIZE, CLARITY_INT_SIZE } from '../../constants';
 import { ClarityType } from '../clarityValue';
 
+const MAX_U128 = new BigNum(2).pow(new BigNum(128)).sub(new BigNum(1));
+const MAX_I128 = new BigNum(2).pow(new BigNum(127)).sub(new BigNum(1));
+const MIN_I128 = new BigNum(-2).pow(new BigNum(127));
+
 interface IntCV {
   readonly type: ClarityType.Int;
   readonly value: BigNum;
 }
 
-const intCV = (value: number | string | Buffer): IntCV => {
-  const bn = new BigNum(value);
-  const twos = bn.toTwos(CLARITY_INT_SIZE);
+function valueToBN(value: unknown, signed: boolean): BigNum {
+  if (typeof value === 'number') {
+    if (!Number.isInteger(value)) {
+      throw new TypeError(`Invalid value. Values of type 'number' must be an integer.`);
+    }
+    return new BigNum(value);
+  }
+  if (typeof value === 'string') {
+    if (value.toLowerCase().startsWith('0x')) {
+      // Convert to buffer then fall through to the buffer condition
+      value = Buffer.from(value.slice(2), 'hex');
+    } else {
+      return new BigNum(value);
+    }
+  }
+  if (typeof value === 'bigint') {
+    return new BigNum(value.toString());
+  }
+  if (value instanceof Uint8Array || Buffer.isBuffer(value)) {
+    let bn: BigNum;
+    if (signed) {
+      if (value.byteLength < CLARITY_INT_BYTE_SIZE) {
+        const paddedBytes = new Uint8Array(CLARITY_INT_BYTE_SIZE);
+        // Extend sign-bit while padding to full byte length.
+        if (value[0] >>> 7) {
+          paddedBytes.fill(0xff);
+        }
+        paddedBytes.set(value, CLARITY_INT_BYTE_SIZE - value.byteLength);
+        bn = new BigNum(paddedBytes, 'be').fromTwos(CLARITY_INT_SIZE);
+      } else {
```

Comment:
@kantai tagged you as a reviewer just to take a look at this sign extension code.